service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -65,4 +75,63 @@ protected static boolean isLastPage(JSONObject response) throws JSONException {
         }
         return result;
     }
+
+    public StatisticalIndicatorDataDimension getTimeperiodDimensionFromIndicatorData(String timePeriodDimensionId,
+            String indicator) {
+        UnsdRequest request = new UnsdRequest(config);
+        request.setIndicator(indicator);
+
+        StatisticalIndicatorDataDimension selector = new StatisticalIndicatorDataDimension(timePeriodDimensionId);
+        Map<Integer, Integer> countOfAreaCodesForYear = new HashMap<>();
+
+        try {
+            while (true) {
+                JSONObject response = JSONHelper.createJSONObject(request.getIndicatorData(null));
+                parseTimePeriod(countOfAreaCodesForYear, response);
+                if (isLastPage(response)) {
+                    break;
+                }
+                request.nextPage();
+            }
+
+            List<IdNamePair> allowedValues = getSortedListOfYearsThatBelongToSeveralGeoAreas(
+                    countOfAreaCodesForYear);
+
+            selector.setAllowedValues(allowedValues);
+            return selector;
+
+        } catch (JSONException ex) {
+            throw new APIException("Error during parsing UNSD data response.", ex);
+        }
+    }
+
+    public static void parseTimePeriod(Map<Integer, Integer> countOfAreaCodesForYear, JSONObject response) {
+        try {
+            JSONArray data = response.getJSONArray(DATA_KEY);
+            for (int i = 0; i < data.length(); i++) {
+                JSONObject o = (JSONObject) data.get(i);
+                Integer year = getTimeperiodYear(o);
+
+                if (countOfAreaCodesForYear.containsKey(year)) {
+                    Integer count = countOfAreaCodesForYear.get(year);
+                    count++;
+                    countOfAreaCodesForYear.put(year, count);
+                } else {
+                    countOfAreaCodesForYear.put(year, 1);
+                }
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing time period selectors for indicator: " + e.getMessage(), e);
+        }
+    }
+    
+    public static List<IdNamePair> getSortedListOfYearsThatBelongToSeveralGeoAreas(
+            Map<Integer, Integer> countOfAreaCodesForYear) {
+        return (List<IdNamePair>) (countOfAreaCodesForYear.entrySet().stream().filter(e -> e.getValue() > 1)
+                .map(e -> new IdNamePair(String.valueOf(e.getKey()), null))).sorted().collect(Collectors.toList());
+    }
+
+    private static Integer getTimeperiodYear(JSONObject o) throws JSONException {
+        return ((Double) o.get(TIME_PERIOD_START_KEY)).intValue();
```

Comment:
This probably works, but there's also `JSONObject#getInt(String key)`

Reply:
Changed code as proposed. Also removed that private method getTimeperiodYear as unnecessary complexity. Logic was originally implemented as private method because of that not so nice type casting and i felt that there was no point for that private method since we get int directly from json object.

Reply:
Changed code as proposed. Also removed that private method getTimeperiodYear as unnecessary complexity. Logic was originally implemented as private method because of that not so nice type casting and i felt that there was no point for that private method since we get int directly from json object.